azure/statestores.bicep

```diff
@@ -26,9 +26,20 @@ param actorStateStore bool = false
 @description('The name of the container to create within the Azure storage account and to reference within the Dapr component.')
 var containerName = context.resource.name
 
+@description('The user-defined tags that will be applied to the resource. Default is null')
+param tags object = {}
+
+@description('The Radius specific tags that will be applied to the resource')
+var radiusTags = {
```

Comment:
Just attempted to deploy this Recipe and got this error message:

```
{
          "code": "RecipeDeploymentFailed",
          "message": "failed to deploy recipe default of type Applications.Dapr/stateStores",
          "details": [
            {
              "code": "DeploymentFailed",
              "message": "At least one resource deployment operation failed. Please see the details for the specific operation that failed.",
              "target": "/planes/radius/local/resourceGroups/default/providers/Microsoft.Resources/deployments/recipe1707868070557672174",
              "details": [
                {
                  "code": "InvalidTagNameCharacters",
                  "message": "The tag names 'radapp.io/application, radapp.io/environment, radapp.io/resource' have reserved characters '\u003c,\u003e,%,\u0026,\\,?,/' or control characters. These characters are only allowed for tags that start with the prefix 'hidden, link'.",
                  "target": "/subscriptions/66d1209e-1382-45d3-99bb-650e6bf63fc0/resourcegroups/aacrawfi/providers/Microsoft.Storage/storageAccounts/recipe2lkqwvspyl24g"
                }
              ]
            }
```

Looking in the Azure portal I also get the same error message: 
![image](https://github.com/radius-project/recipes/assets/54363786/5cbbf0b0-eb25-49eb-a69f-e9061c0b9789)


We should switch to using another character, maybe `-`? (`radapp.io-application`)

cc/ @rynowak

Reply:
@AaronCrawfis I've changed the character now to a hyphen :)

Question - Do the tests just work for the local recipes? If so, could we introduce tests to test the Azure/AWS recipes as part of the workflow? Initially I thought using ephemeral environments for each cloud to do this, but not 100% on this.